src/hal/hal_i2c_microSpecific.h

```diff
@@ -0,0 +1,33 @@
+#ifndef HAL_I2C_MICRO_SPECIFIC_H
+#define HAL_I2C_MICRO_SPECIFIC_H
+
+#include "hal.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
```

Comment:
These typedefs can be shared between different microcontrollers, I think it is fair to say many others will support the same basic functionality shown here. I think these should live in a `hal_i2c.h` in the hal repo and not be microspecific.